lms/djangoapps/grades/signals/handlers.py

```diff
@@ -131,3 +131,14 @@ def enqueue_grade_update(sender, **kwargs):  # pylint: disable=unused-argument
             kwargs.get('only_if_higher'),
         )
     )
+
+
+@receiver(COURSE_GRADE_UPDATE_REQUESTED)
+def enqueue_course_update(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Handles the COURSE_GRADE_UPDATE_REQUESTED signal by enqueueing a course update operation to occur asynchronously.
+    """
+    if sender is recalculate_subsection_grade:  # We're already in a async worker, just do the task
```

Comment:
When would we need the bulk update? The course grade code is already set up to (on first access) create all the subsection grades if they don't exist. The main use case I've been working with is "user submits problem answer, which triggers a grade update". Do we expect these operations to come in quickly enough that thrashing may be a problem? Or is there another use case I'm overlooking?

Agreed on the `isinstance` suggestion, thanks!